libbpf-rs/src/object.rs

```diff
@@ -152,17 +140,44 @@ impl Default for ObjectBuilder {
     }
 }
 
-/// Represents a BPF object file. An object may contain zero or more
-/// [`Program`]s and [`Map`]s.
-pub struct Object {
+fn find_map_in_object(
+    obj: *const libbpf_sys::bpf_object,
+    name: &str,
+) -> Result<Option<*mut libbpf_sys::bpf_map>> {
+    let c_name = util::str_to_cstring(name)?;
+    let ptr = unsafe { libbpf_sys::bpf_object__find_map_by_name(obj, c_name.as_ptr()) };
+    if ptr.is_null() {
```

Comment:
Rust offtopic: I wonder if having generic Null->Option generic conversion routine would be possible?